### PR TITLE
Path generation

### PR DESCRIPTION
**Pull request by johnnyb8**

### CODE REVIEW

**Review by s-axford — APPROVED** [review]
Looks good to me, although we may want to consider widening the paths

File: stock-filters/ChirpVillage/Block.py [hunk 1]
```diff
@@ -23,3 +23,6 @@ def __init__(self, x, z, height, type):
         self.height = height
         self.type = type
         self.biome_id = -1  # undefined by default, set in calculateBiomeMap of BlockUtils
+    
+    def set_type(self, new_type):
```
Comment (JeremyButt) [inline stock-filters/ChirpVillage/Block.py]: why is this needed? nothing is private. we can modify the type member directly.

File: stock-filters/ChirpVillage/PathGeneration.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+from Mocks.Box import BoundingBox
+from Mocks.Level import Level
```
Comment (JeremyButt) [inline stock-filters/ChirpVillage/PathGeneration.py]: These should be moved into the `if __name__ == "__main__":` block as the mocks should only be loaded when testing.

File: stock-filters/ChirpVillage/PathGeneration.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    # Returns whether the cell is within the grid
+    def block_in_grid(self, block):
+        return 0 <= block[0] < self.surface.x_length and 0 <= block[1] < self.surface.z_length
+
+    def get_block_neighbors(self, block):
+        neighbors = [self.add(block, connection) for connection in self.directions]
+        neighbors = filter(self.block_in_grid, neighbors)
+        neighbors = filter(self.block_is_free, neighbors)
+        return neighbors
```
Comment (JeremyButt) [inline stock-filters/ChirpVillage/PathGeneration.py:44]: This is a really fucking clean function! Good job! 🧼

File: stock-filters/ChirpVillage/Surface.py [hunk 4]
```diff
@@ -22,6 +22,9 @@ def __init__(self, level, box):
         self.surface_map = self.init_surface_map(level)
         self.door_blocks = []
 
+    def set_surface_map(self, surface_map):
```
Comment (JeremyButt) [inline stock-filters/ChirpVillage/Surface.py]: Why is this needed? its public and can be modified and set directly

File: stock-filters/ChirpVillage/Surface.py [hunk 5]
```diff
@@ -93,8 +96,9 @@ def visualize(self):
...
                 row.append("_" if self.surface_map[i][j].type == Block.YARD else
                            "#" if self.surface_map[i][j].type == Block.BUILDING else
-                           "|" if self.surface_map[i][j].type == Block.DOOR else
+                           "X" if self.surface_map[i][j].type == Block.DOOR else
+                           "@" if self.surface_map[i][j].type == Block.PATH else
                            " ")
             row.append("|")
             print("  ".join(row))
-        print("| " + "".join([" + " for _ in range(self.x_length)]) + " |")
+        print("| " + "".join([" + " for _ in range(self.x_length)]) + " |")
```
Comment (JeremyButt) [inline stock-filters/ChirpVillage/Surface.py]: Newline

**Comment by JeremyButt** [comment]
Looks good! couple of changes requested! other than that really good implementation! 👍

File: stock-filters/ChirpVillage/Block.py (hunk 1 above)
Comment (johnnyb8) [reply stock-filters/ChirpVillage/Block.py]: meant to take this out

File: stock-filters/ChirpVillage/Surface.py (hunk 4 above)
Comment (johnnyb8) [reply stock-filters/ChirpVillage/Surface.py]: meant to take this out

**Review by TimothyMather — APPROVED** [review]

**Review by JeremyButt — APPROVED** [review]
